File: Java/dependency-injection/changing-behavior-through-composition/src/main/java/com/github/grzesiek_galezowski/ChangingBehaviorThroughComposition/SwitchCriterias/MetDuringTheDayCriteria.java
package com.github.grzesiek_galezowski.ChangingBehaviorThroughComposition.SwitchCriterias;

import com.github.grzesiek_galezowski.ChangingBehaviorThroughComposition.SwitchCriteria;

import java.time.LocalTime;

public class MetDuringTheDayCriteria implements SwitchCriteria
  {
    public boolean isNotMet()
    {
      //buggy implementation. Nevermind...
      return isNight();
    }

    private static boolean isNight()
    {
      int currentHour = LocalTime.now().getHour();
      return currentHour > 22 && currentHour < 8;
    }
  }

File: CSharp/DependencyInjection/ChangingBehaviorThroughComposition/SwitchCriterias/MetDuringTheDayCriteria.cs
using System;

namespace ChangingBehaviorThroughComposition
{
  public class MetDuringTheDayCriteria : SwitchCriteria
  {
    public bool IsNotMet()
    {
      //buggy implementation. Nevermind...
      return IsNight();
    }

    private static bool IsNight()
    {
      var currentHour = DateTime.Now.Hour;
      return currentHour > 22 && currentHour < 8;
    }
  }
}